File: AFMobile/AFAndroid/app/src/main/java/cz/cvut/fel/matyapav/afandroid/enums/SupportedWidgets.java
package cz.cvut.fel.matyapav.afandroid.enums;

/**
 * Enumeration of supported widget types
 *
 * @author Pavel Matyáš (matyapav@fel.cvut.cz).
 *@since 1.0.0..
 */
public enum SupportedWidgets {

    TEXTFIELD("TEXTFIELD"),
    PASSWORD("PASSWORD"),
    NUMBERFIELD("NUMBERFIELD"),
    NUMBERDOUBLEFIELD("NUMBERDOUBLEFIELD"),
    CALENDAR("CALENDAR"),
    OPTION("OPTION"),
    DROPDOWNMENU("DROPDOWNMENU"),
    CHECKBOX("CHECKBOX");

    private String widgetName;

    SupportedWidgets(String widgetType) {
        this.widgetName = widgetType;
    }

    public String getWidgetName() {
        return widgetName;
    }
}

File: AFMobile/AFWinPhone/AFWinPhone/enums/SupportedWidgets.cs
using System;

namespace AFWinPhone.enums
{
    public sealed class SupportedWidgets
    {
        public static readonly SupportedWidgets TEXTFIELD = new SupportedWidgets("TEXTFIELD");
        public static readonly SupportedWidgets PASSWORD = new SupportedWidgets("PASSWORD");
        public static readonly SupportedWidgets NUMBERFIELD = new SupportedWidgets("NUMBERFIELD");
        public static readonly SupportedWidgets NUMBERDOUBLEFIELD = new SupportedWidgets("NUMBERDOUBLEFIELD");
        public static readonly SupportedWidgets CALENDAR = new SupportedWidgets("CALENDAR");
        public static readonly SupportedWidgets OPTION = new SupportedWidgets("OPTION");
        public static readonly SupportedWidgets DROPDOWNMENU = new SupportedWidgets("DROPDOWNMENU");
        public static readonly SupportedWidgets CHECKBOX = new SupportedWidgets("CHECKBOX");

        private String widgetName;

        SupportedWidgets(String widgetType)
        {
            this.widgetName = widgetType;
        }

        public String getWidgetName()
        {
            return widgetName;
        }
    }
}
